### ISSUE TITLE
`MagicNumber` rule not working with cli on 2.x

### ISSUE THREAD

**Issue by marschwar** [detekt/detekt#8667]
## Expected Behavior
Assuming the following code, a violation of the `MagicNumber` rule is detected.
```kotlin
class Consumer {
    val foo : Int = 567
}

```

## Observed Behavior
When running the detekt using the cli of `2.0.0-alpha.0`, no violation is detected. 

Using the cli in version `1.23.8` or analyzing the source with the 2.0.0 gradle plugin, the violation is detected.

## Steps to Reproduce
* download and unpack https://github.com/detekt/detekt/releases/download/v1.23.8/detekt-cli-1.23.8.zip
* download and unpack https://github.com/detekt/detekt/releases/download/v2.0.0-alpha.0/detekt-cli-2.0.0-alpha.0.zip
* create a kotlin source file from above
* run ` $HOME/Downloads/detekt-cli-1.23.8/bin/detekt-cli  --all-rules -i app/src `
* run `$HOME/Downloads/detekt-cli-2.0.0-alpha.0/bin/detekt-cli  --all-rules -i app/src `

**Comment by marschwar** [comment]
Looked into this a little more. Something is off here. I will keep looking.

**2.0.0**
```
$> java -jar detekt-cli-2.0.0-alpha.0/lib/detekt-cli-2.0.0-alpha.0-all.jar --debug --all-rules 
Loaded extensions: 
    
Registered rule sets: 
    dev.detekt.rules.complexity.ComplexityProvider
Registered rule sets: 
    dev.detekt.rules.complexity.ComplexityProvider

Loaded extensions: 
    
Loaded extensions: 
    dev.detekt.core.baseline.BaselineResultMapping
Loaded extensions: 
    dev.detekt.report.md.MdOutputReport
Loaded extensions: 
    dev.detekt.core.reporting.console.LiteIssuesReport
...
```

**1.23.8**
```
java -jar detekt-cli-1.23.8/lib/detekt-cli-1.23.8-all.jar --debug --all-rules 
Loaded extensions: 
    
Registered rule sets: 
    io.gitlab.arturbosch.detekt.rules.bugs.PotentialBugProvider@9ef8eb7
    io.gitlab.arturbosch.detekt.rules.complexity.ComplexityProvider@34cdeda2
    io.gitlab.arturbosch.detekt.rules.coroutines.CoroutinesProvider@6ee660fb
    io.gitlab.arturbosch.detekt.rules.documentation.CommentSmellProvider@305a0c5f
    io.gitlab.arturbosch.detekt.rules.empty.EmptyCodeProvider@4535b6d5
    io.gitlab.arturbosch.detekt.rules.exceptions.ExceptionsProvider@1ecee32c
    io.gitlab.arturbosch.detekt.rules.naming.NamingProvider@4372b9b6
    io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider@232a7d73
    io.gitlab.arturbosch.detekt.rules.style.StyleGuideProvider@4b41e4dd
Loaded extensions: 
    io.gitlab.arturbosch.detekt.rules.documentation.LicenceHeaderLoaderExtension@4c18621b
    io.github.detekt.metrics.processors.ProjectSLOCProcessor@39c385d6
    io.github.detekt.metrics.processors.ProjectLOCProcessor@1cec219f
    io.github.detekt.metrics.processors.ProjectCLOCProcessor@692e028d
    io.github.detekt.metrics.processors.ProjectLLOCProcessor@6629ad09
    io.github.detekt.metrics.processors.ProjectCognitiveComplexityProcessor@43e1692f
    io.github.detekt.metrics.processors.ProjectComplexityProcessor@3e7c4815
    io.github.detekt.metrics.processors.PropertyCountProcessor@27df0f3d
    io.github.detekt.metrics.processors.FunctionCountProcessor@c35af2a
    io.github.detekt.metrics.processors.ClassCountProcessor@44550792
    io.github.detekt.metrics.processors.PackageCountProcessor@1835d3ed
    io.github.detekt.metrics.processors.KtFileCountProcessor@474c9131
Registered rule sets: 
    io.gitlab.arturbosch.detekt.rules.bugs.PotentialBugProvider@5c8e67b9
    io.gitlab.arturbosch.detekt.rules.complexity.ComplexityProvider@260a3a5e
    io.gitlab.arturbosch.detekt.rules.coroutines.CoroutinesProvider@49206065
    io.gitlab.arturbosch.detekt.rules.documentation.CommentSmellProvider@3c0bbc9f
    io.gitlab.arturbosch.detekt.rules.empty.EmptyCodeProvider@1317b708
    io.gitlab.arturbosch.detekt.rules.exceptions.ExceptionsProvider@6438a7fe
    io.gitlab.arturbosch.detekt.rules.naming.NamingProvider@2e51d054
    io.gitlab.arturbosch.detekt.rules.performance.PerformanceProvider@4f5f6e45
    io.gitlab.arturbosch.detekt.rules.style.StyleGuideProvider@615e3f51
...
Loaded extensions: 
    io.gitlab.arturbosch.detekt.core.baseline.BaselineResultMapping@cc9ef8d
Loaded extensions: 
    io.github.detekt.report.sarif.SarifOutputReport@426131d7
    io.github.detekt.report.xml.XmlOutputReport@5d9618f2
    io.github.detekt.report.txt.TxtOutputReport@7c7e73c5
    io.github.detekt.report.html.HtmlOutputReport@13cc0b90
    io.github.detekt.report.md.MdOutputReport@58687fb7
Loaded extensions: 
    io.gitlab.arturbosch.detekt.core.reporting.console.LiteFindingsReport@7f8a9454
...
```

**Comment by marschwar** [comment]
The file `META-INF/services/dev.detekt.api.RuleSetProvider` no longer contains all providers but only `dev.detekt.rules.complexity.ComplexityProvider`

**Comment by marschwar** [comment]
From the shadow jar docs: https://gradleup.com/shadow/configuration/merging/#handling-duplicates-strategy

> NOTE: The duplicatesStrategy takes precedence over transforming and relocating. If you mix the usages of duplicatesStrategy and [ResourceTransformer](https://gradleup.com/shadow/api/shadow/com.github.jengelman.gradle.plugins.shadow.transformers/-resource-transformer/) like below:
> 
> 
> ```kotlin
> tasks.shadowJar {
>   duplicatesStrategy = DuplicatesStrategy.EXCLUDE // The default strategy.
>   mergeServiceFiles()
> }
> ```
> 
> The [ResourceTransformer](https://gradleup.com/shadow/api/shadow/com.github.jengelman.gradle.plugins.shadow.transformers/-resource-transformer/)s like [ServiceFileTransformer](https://gradleup.com/shadow/api/shadow/com.github.jengelman.gradle.plugins.shadow.transformers/-service-file-transformer/) will not work as expected as the duplicate resource files fed for them are excluded beforehand. However, this behavior might be what you expected for duplicate foo/bar files, preventing them from being included.

**Comment by marschwar** [comment]
Most likely caused by https://github.com/GradleUp/shadow/releases/tag/9.0.1 

> Changed
> 
> - Improve the error message for empty mainClassName. (https://github.com/GradleUp/shadow/pull/1601)
> - Default duplicatesStrategy back to EXCLUDE. (https://github.com/GradleUp/shadow/pull/1617)
>   - This strategy is consistent with 8.x series behavior, which is more compatible for most users upgrading.
>   - For most ResourceTransformer users, you need to override the strategy to INCLUDE to make them work.
>   - Strongly suggest declaring the duplicatesStrategy explicitly in your ShadowJar configuration to avoid confusion.
>   - See more details about the strategies at [Handling Duplicates Strategy](https://gradleup.com/shadow/configuration/merging/#handling-duplicates-strategy).

**Comment by BraisGabin** [comment]
@Goooler raised 2 PRs related with that and they are still pending to merge: https://github.com/detekt/detekt/pulls/Goooler maybe that helps.
